amqp-client/src/test/java/zipkin2/reporter/amqp/RabbitMQSenderRule.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter.amqp;
+
+import java.time.Duration;
+import org.junit.AssumptionViolatedException;
+import org.junit.ClassRule;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import zipkin2.CheckResult;
+
+import static zipkin2.Call.propagateIfFatal;
+
+/** This should be used as a {@link ClassRule} as it takes a very long time to start-up. */
+class RabbitMQSenderRule extends ExternalResource {
+  static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQSenderRule.class);
+  static final String IMAGE = "rabbitmq:3.7-management-alpine";
+  static final String QUEUE = "zipkin-test1";
+  static final int RABBIT_PORT = 5672;
+
+  static final class RabbitMQContainer extends GenericContainer<RabbitMQContainer> {
+    RabbitMQContainer(String image) {
+      super(image);
+      addExposedPorts(RABBIT_PORT);
+      this.waitStrategy = Wait.forLogMessage(".*Server startup complete.*", 1)
+          .withStartupTimeout(Duration.ofSeconds(60));
+    }
+  }
+
+  RabbitMQContainer container;
+
+  @Override protected void before() {
+    if ("true".equals(System.getProperty("docker.skip"))) {
+      throw new AssumptionViolatedException("Skipping startup of docker " + IMAGE);
+    }
+
+    try {
+      LOGGER.info("Starting docker image " + IMAGE);
+      container = new RabbitMQContainer(IMAGE);
```

Comment:
FYI we no longer perform _any_ side effect in constructors, so it is safe to make it final